Src/WitsmlExplorer.Api/Services/CredentialsCache.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.Caching;
+
+using Microsoft.Extensions.Logging;
+namespace WitsmlExplorer.Api.Services
+{
+
+    public interface ICredentialsCache
+    {
+        void Set(string cacheId, string encryptedCredentials, CacheItemPolicy policy);
+        public string Get(string cacheId);
+        public long Count();
+        public void RefreshSession(string cacheId, string encryptedCredentials);
+        public void RemoveAllClientCredentials(string clientId);
+    }
+
+    public class CredentialsCache : ICredentialsCache
+    {
+        private readonly ObjectCache _cache = MemoryCache.Default;
+        private readonly ILogger<CredentialsCache> _logger;
+
+        public CredentialsCache(ILogger<CredentialsCache> logger)
+        {
+            _logger = logger;
+        }
+
+        public void Set(string cacheId, string encryptedCredentials, CacheItemPolicy policy)
```

Comment:
Would it be better to pass the "keep" boolean here so that the cache can decide the policy based on it?

Reply:
Not sure, it seems to be a little tight connected to the format of the endpoint query then? 
But we could improve by accepting TTL as a number and then initialize the policy from within the CredentialsCache.
I will implement that for now, what do you think?

Reply:
_Not sure, it seems to be a little tight connected to the format of the endpoint query then?_
Well I looked at it in a way that in practice we use two cache policies, and which police is used is based on the _keep_ variable.  I think it is barely important given that we construct the policy in only one place (CredentialsService), and should rather be decided with regards to being easy to adjust the policies in code, such as for the SlidingExpiration suggestion.

Reply:
The `SlidingExpiration` was great. This has been implemented now 👍  I think I will keep the `ttl` approach for now because it seems to be a little more generic and opening up for more than two states for the cache items through the same method.